tests/resources/promise-rejection-events.js

```diff
@@ -446,14 +446,14 @@ async_test(function(t) {
    'all inside a setTimeout');
 
 //
-// Positive unhandledrejection/rejectionhandled tests with delayed attachment
+// Negative unhandledrejection/rejectionhandled tests with task-delayed attachment
 //
 
 async_test(function(t) {
   var e = new Error();
   var p;
 
-  onUnhandledSucceed(t, e, function() { return p; });
+  onUnhandledFail(t, function() { return p; });
```

Comment:
actually, you're right.

the tests then pass on document but not on worker, since the
postMessageTask() thing

I've updated the CL with tests + implementation

if you update your spec text, and import my tests, we should be good to go

On Fri, Oct 2, 2015 at 2:10 PM Domenic Denicola <notifications@github.com>
wrote:

> In tests/resources/promise-rejection-events.js
> <https://github.com/domenic/unhandled-rejections-browser-spec/pull/13#discussion_r41014865>
> :
>
> >  //
> >
> >  async_test(function(t) {
> >    var e = new Error();
> >    var p;
> >
> > -  onUnhandledSucceed(t, e, function() { return p; });
> > +  onUnhandledFail(t, function() { return p; });
>
> Why? I thought that was the entire content of the discussion in #2
> <https://github.com/domenic/unhandled-rejections-browser-spec/issues/2>.
> We should not send events if the promise has been handled by the time we
> *would* send the events.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/domenic/unhandled-rejections-browser-spec/pull/13/files#r41014865>
> .
>